app/services/data_exports/destroy_service.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module DataExports
+  # Service used to Destroy Data Exports
+  class DestroyService < BaseService
+    def initialize(data_export, user = nil)
+      super(user, {})
+      @data_export = data_export
+    end
+
+    def execute
+      authorize! @data_export, to: :destroy?
+
+      @data_export.file.blank? || @data_export.file.purge
```

Comment:
**nitpick (blocking):** Can you remove `acts_as_paranoid` and `deleted_at` column from `DataExport` model as we don't want these to be soft deletable. Can you also remove this line and instead update `has_one_attached` to include `dependant: :purge_later` so we don't have to wait for the file to be purged before returning from here. https://api.rubyonrails.org/v7.1.3.2/classes/ActiveStorage/Attached/Model.html#method-i-has_one_attached

Reply:
Fixed in 578a177b52473dcc8f9226cfdd4933cc1b96745e